src/frontends/src/json_util.hpp

```diff
@@ -0,0 +1,203 @@
+#include <rapidjson/document.h>
+#include <clipper/datatypes.hpp>
+#include <stdexcept>
+
+using clipper::Input;
+using clipper::InputType;
+using clipper::Output;
+using clipper::OutputType;
+using clipper::VersionedModelId;
+using rapidjson::Type;
+
+static std::vector<std::string> kTypeNames = {
+    "Null", "False", "True", "Object", "Array", "String", "Number"};
+
+// RapidJSON value types
+/*
+enum Type {
+  kNullType = 0,      // null
+  kFalseType = 1,     // false
+  kTrueType = 2,      // true
+  kObjectType = 3,    // object
+  kArrayType = 4,     // array
+  kStringType = 5,    // string
+  kNumberType = 6     // number
+}; */
+
+namespace clipper_json {
+
+class json_parse_error : public std::runtime_error {
+ public:
+  json_parse_error(const std::string& what) : std::runtime_error(what) {}
+  ~json_parse_error() throw(){};
+};
+
+class json_semantic_error : public std::runtime_error {
+ public:
+  json_semantic_error(const std::string& what) : std::runtime_error(what) {}
+  ~json_semantic_error() throw(){};
+};
+
+/* Check for matching types else throw exception */
+void check_type(rapidjson::Document& d, const char* key_name,
+                Type expected_type) {
+  if (!d.HasMember(key_name)) {
+    throw json_semantic_error("JSON object does not have required key: " +
+                              std::string(key_name));
+  }
+  rapidjson::Value& val = d[key_name];
+  if (val.GetType() != expected_type) {
+    throw json_semantic_error("Type mismatch! JSON key " +
+                              std::string(key_name) + " expected type " +
+                              kTypeNames[expected_type] + "but found type " +
+                              kTypeNames[val.GetType()]);
+  }
+}
+
+/* Getters with error handling for double, float, long, int, string */
+double getDouble(rapidjson::Document& d, const char* key_name) {
```

Comment:
The five non-array get functions all follow the same pattern. Wanted to get your input on the error message.